static/app/components/modals/metricWidgetViewerModal.tsx

```diff
@@ -0,0 +1,319 @@
+        </Body>
+        <Footer>
+          <ButtonBar gap={1}>
+            <Button
+              onClick={() => {
+                closeModal();
+                router.push(
+                  getDdmUrl(organization.slug, {
+                    widgets: [metricWidgetQueryParams],
+                    project: selection.projects,
+                    ...selection.datetime,
+                    environment: selection.environments,
+                  })
+                );
+              }}
+            >
+              {t('Open in Metrics')}
+            </Button>
+            <Button
+              onClick={() => {
+                closeModal();
+              }}
+            >
+              {t('Close')}
+            </Button>
+            <Button
+              color="primary"
+              priority="primary"
+              onClick={() => {
+                handleSubmit();
+                closeModal();
+              }}
+            >
+              {t('Save changes')}
+            </Button>
+          </ButtonBar>
+        </Footer>
+      </OrganizationContext.Provider>
+    </Fragment>
+  );
+}
+export function ContextMenu({
+  metricsQuery,
+}: {
+  metricsQuery: MetricsQuery;
+}) {
+  const organization = useOrganization();
+  const router = useRouter();
+
+  const createAlert = useMemo(
+    () => getCreateAlert(organization, metricsQuery),
+    [metricsQuery, organization]
+  );
+
+  const items = useMemo<MenuItemProps[]>(() => {
+    const customMetric = !isCustomMetric({mri: metricsQuery.mri});
+    const addAlertItem = {
+      leadingItems: [<IconSiren key="icon" />],
+      key: 'add-alert',
+      label: t('Create Alert'),
+      disabled: !createAlert,
+      onAction: () => {
+        trackAnalytics('ddm.create-alert', {
+          organization,
+          source: 'widget',
+        });
+        Sentry.metrics.increment('ddm.dashboards.widget.alert');
+        createAlert?.();
+      },
+    };
+    const settingsItem = {
+      leadingItems: [<IconSettings key="icon" />],
+      key: 'settings',
+      label: t('Metric Settings'),
+      disabled: !customMetric,
+      onAction: () => {
+        trackAnalytics('ddm.widget.settings', {
+          organization,
+        });
+        Sentry.metrics.increment('ddm.dashboards.widget.settings');
+        navigateTo(
+          `/settings/projects/:projectId/metrics/${encodeURIComponent(metricsQuery.mri)}`,
+          router
+        );
+      },
+    };
+
+    return customMetric ? [addAlertItem, settingsItem] : [addAlertItem];
+  }, [createAlert, metricsQuery.mri, organization, router]);
+
+  return (
+    <DropdownMenu
+      items={items}
+      triggerProps={{
+        'aria-label': t('Widget actions'),
+        size: 'md',
+        showChevron: false,
+        icon: <IconEllipsis direction="down" size="sm" />,
+      }}
+      position="bottom-end"
+    />
+  );
+}
+
+export default withPageFilters(MetricWidgetViewerModal);
```

Comment:
why not use `usePageFilters`? copy pasta? 🙃

Reply:
Yep